swarm/api/http/server.go

```diff
@@ -975,7 +976,38 @@ func (s *Server) HandleGetFile(w http.ResponseWriter, r *Request) {
 	}
 
 	w.Header().Set("Content-Type", contentType)
-	http.ServeContent(w, &r.Request, "", time.Now(), reader)
+	http.ServeContent(w, &r.Request, "", time.Now(), newBufferedReadSeeker(reader, getFileBufferSize))
+}
+
+// The size of buffer used for bufio.Reader on LazyChunkReader passed to
+// http.ServeContent in HandleGetFile.
+// Warning: This value influences the number of chunk requests and chunker join goroutines
+// per file request.
+// Recommended value is 4 times the io.Copy default buffer value which is 32kB.
```

Comment:
Why just 4*? Based on the branching factor we have, I think we can go much higher, like 32* or 64*, no?

Reply:
While testing on open cluster, raising over 4 did not improve download speed. It was saturated after 4 (32 goroutines), with 100 download was practically impossible as many chunk requests were timing out.

Reply:
Hmm... ok.

Reply:
This feels quite inefficient... if our requests are for single chunks, I don't think we will manage to make it much more efficient. We should probably batch the requests for chunks, similarly to the Syncing protocol.

This is just my intuition, I might be completely wrong here.

Reply:
Yes, some sort of concurrent chunk retrieval limit is required.

Reply:
a year ago i had an issue to package requests but i dont think it is worth it. more likely it is the received chunk deliveries is what clogs incoming processing. we should look into this more thoroughly with metrics after chunk reffactor